typing/ikinds/ldd_hybrid_opt.ml

```diff
@@ -238,76 +277,84 @@ module Make (V : ORDERED) = struct
       else node nb.v (meet a nb.lo) (meet a nb.hi)
 
   and meet_with_leaf (leaf_a : node) (other : node) =
-    let leaf_val = leaf_value leaf_a in
-    if C.equal leaf_val C.top
+    let rec aux (leaf_a : Axis_lattice.t) (other : node) =
+      if is_leaf other
+      then leaf (Axis_lattice.meet leaf_a (Unsafe.leaf_value other))
+      else
+        let nb = Unsafe.node_block other in
+        let lo' = aux leaf_a nb.lo in
+        let hi' = aux leaf_a nb.hi in
+        if lo' == nb.lo && hi' == nb.hi then other else node nb.v lo' hi'
+    in
+    let leaf_val = Unsafe.leaf_value leaf_a in
+    if Axis_lattice.equal leaf_val Axis_lattice.top
     then other
-    else if C.equal leaf_val C.bot
+    else if Axis_lattice.equal leaf_val Axis_lattice.bot
     then bot
-    else meet_with_leaf_aux leaf_val other
-
-  and meet_with_leaf_aux (leaf_a : C.t) (other : node) =
-    if is_leaf other
-    then leaf (C.meet leaf_a (leaf_value other))
-    else
-      let nb = node_block other in
-      let lo' = meet_with_leaf_aux leaf_a nb.lo in
-      let hi' = meet_with_leaf_aux leaf_a nb.hi in
-      if lo' == nb.lo && hi' == nb.hi then other else node nb.v lo' hi'
+    else aux leaf_val other
 
   (* --------- public constructors --------- *)
-  let[@inline] const (c : C.t) = leaf c
+  let[@inline] const (c : Axis_lattice.t) = leaf c
 
   let[@inline] mk_var (v : var) : node = v.var_node
 
   let rigid (name : V.t) = Var.make_rigid ~name ()
 
   let new_var () = Var.make_var ()
 
-  (* --------- restrictions (x ← ⊥ / ⊤) (no memoization) --------- *)
-  let rec restrict0 (x : var) (w : node) : node =
+  (* --------- assignments (x ← ⊥ / ⊤) ---------
+     These are cofactors: assign the given variable to bottom/top and
+     simplify the diagram accordingly. *)
```

Comment:
I'd prefer leaving individual doc comments on each function rather than one comment above the two. Although this structure makes more sense when authoring a file and reading it as a whole, this comment can be hard to find if, say, someone saw a usage of `assign_top` and is trying to figure out what it does.

Reply:
ex:
```
(** [assign_bot ~var w] substitutes [var] with the bottom value within node [w], simplifying the diagram accordingly. *)
```